core/src/value/primitive.rs

```diff
@@ -2540,6 +2565,15 @@ mod tests {
         assert_eq!(value.to_str(), "10\\11\\12",);
     }
 
+    #[test]
+    fn primitive_value_to_clean_str(){
+        let value = PrimitiveValue::from("1.2.345\0".to_string());
+        assert_eq!(&value.to_clean_str(), "1.2.345");
+
+        let value = dicom_value!(Strs, ["ONE", "TWO", "THREE", "SIX "]);
+        assert_eq!(&value.to_clean_str(), "ONE\\TWO\\THREE\\SIX");
+    }
```

Comment:
These cases are good, and can be copied over to the example doctest above. On the other hand, there is an opportunity to improve coverage here. Let's also add a case with leading spaces (which should be preserved) and another one for some other binary representation (which should still work even when the trimming is unnecessary).

Reply:
Thank you for the feedback! 
I've made some changes to the code and even added more tests with leading spaces. Could you provide, if possible, some test examples with DicomValue and DataElement? Thanks!

Reply:
Sure, for `DicomValue` and `DataElement`, it is sufficient to replicate the primitive values already tested in the `primitive` module and encapsulate them into the other types. `From` implementations are available for `DicomValue`, and constructors exist for both types.

For instance, the following attribute:
``` 
(0028,3002) LUTDescriptor                       US (3,  6 bytes): [256, 0, 16]
```

Would be represented as a data element like this:

```rust
let p = dicom_value!(U16, [256, 0, 16]);
let val = DicomValue::new(p);
let element = DataElement::new(Tag(0x0028, 0x3002), VR::US, val);
assert_eq!(element.to_clean_str()?, "256\\0\\16");
```